### DISCUSSION TITLE
How to choose default date/time format?

### DISCUSSION THREAD

**Discussion by dmytrokyrychuk** [freeplane/freeplane#864] [Help] [answered]
When I am typing a date in ISO 8601 format, it is changed to another format when I finish editing the node. In order to restore the ISO 8601 format, I have to change the format in the Manage Time dialog:

https://user-images.githubusercontent.com/699961/205616317-3c76c3c9-6c58-4185-ad8f-aa7d73e73e94.mov

Is there any way I can avoid the automatic date format conversion and keep the date in the ISO 8601 format? I have searched through the Preferences without success.

I am running Freeplane v1.10.5

**Comment by EdoFro** [comment]
In the **user directory** (menu Tools->Open user directory) are two xml files that you can edit to control how the automatic formatting is applied:

- format.xml
- scanner.xml

I don't remember right now which one do you have to edit and how. I did it once and it worked. That's all I remember.

I hope it helps

edo

**Reply by dmytrokyrychuk** [reply]
After @macmarrum's response I changed `Preferences…->Behaviour->Data formatting and parsing->Standard date format` and `Preferences…->Behaviour->Data formatting and parsing->Standard date-time format`, which caused the following lines to be changed in `auto.properties`:

```
date_format=yyyy-MM-dd
datetime_format=yyyy-MM-dd HH\:mm
```

I attempted to change formats in `format.xml`, but I did not notice any changes even after restarting Freeplane. But I am happy with the current solution. Thanks, @EdoFro!

**Comment by macmarrum** [comment, marked as answer]
> Is there any way I can avoid the automatic date format conversion and keep the date in the ISO 8601 format?

There are two points here:

1. Avoid automatic date format conversion → this can be done by disabling format recognition, so that the value that you type stays as text: `Preferences…->Behaviour->Data formatting and parsing->Recognize input of numbers and date-time`
2. Keep the date in the ISO 8601 format → in addition to the solution provided by EdoFro, you can change `Preferences…->Behaviour->Data formatting and parsing->Standard date format` and set it to `yyyy-MM-dd`; this will be the default in Freeplane, **if format recognition is enabled**

**Reply by dmytrokyrychuk** [reply]
Thanks so much. I feel bad for asking this question knowing that there indeed was an option I could have changed myself in the Preferences window :facepalm:

**Comment by EdoFro** [comment]
I deleted my second post because it was just a way around and not a direct answer to the question.
